### PR TITLE
feat(uat): implement scenario outline GGMQ-1-T27

### PR DESCRIPTION
**Pull request by bgklika**
**Issue #, if available:**
Implement GGMQ scenario based on GGAD-1-T27

**Description of changes:**
- Add GGMQ-1-T27 scenario outline
- Move list of addresses of brokers from MqttControlSteps to MqttBrokers
- Add BrokerCertificateSteps with implementation of certificate getting and comparision
- In addition: update readme with description of tags

**Why is this change necessary:**
Implement replacement of GGAD-1-T27 scenario outline

**How was this change tested:**
Scenarios have been run on Linux for 4 clients (later reduced to 1) and Windows (one client) for both MQTT v3 and v5.

**Test results:**
Linux:
```
[INFO ] 2023-07-25 22:46:59.173 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v3-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
[INFO ] 2023-07-25 22:46:59.173 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v5-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
```

Windows:
```
[INFO ] 2023-07-25 23:22:47.231 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v3-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
[INFO ] 2023-07-25 23:22:47.231 [main] StepTrackingReporting - Passed: 'GGMQ-1-T27-v5-sdk-java: As a customer, my Greengrass-issued certificate does not rotate when mqtt reconnects'
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature [hunk 1]
```diff
@@ -1740,6 +1740,102 @@ Feature: GGMQ-1
...
+    #  sometimes detecting IP and updating cert takes >40 seconds
+    And I wait 65 seconds
+
+    And I discover core device broker as "default_broker" from "clientDeviceTest" in OTF
+    And I connect device "clientDeviceTest" on <agent> to "default_broker" using mqtt "<mqtt-v>"
+
+    Then I retrieve the certificate of broker "default_broker" and store as "BROKER_CERTIFICATE_BEFORE_DISCONNECT"
+
+    When I set device mqtt connectivity to offline
+    And the greengrass log on the device contains the line "com.aws.greengrass.mqttclient.AwsIotMqtt5Client: Failed to connect to AWS IoT Core" at least 3 times within 2 minutes
```
Comment (MikeDombo) [inline uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: why do you want this 3 times? Why isn't 1 enough?

Comment (bgklika) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: At first time 1 is enough. Usually Nucleus makes 3 attempts in the short period of time like:
```
2023-07-25T20:43:30.342Z [ERROR] (Thread-10) com.aws.greengrass.mqttclient.AwsIotMqtt5Client: Failed to connect to AWS IoT Core. {clientId=gg-5e37c73c40370bb47008-ggc-thing, error=socket operation timed out.}
2023-07-25T20:43:34.858Z [ERROR] (Thread-10) com.aws.greengrass.mqttclient.AwsIotMqtt5Client: Failed to connect to AWS IoT Core. {clientId=gg-5e37c73c40370bb47008-ggc-thing, error=socket operation timed out.}
2023-07-25T20:43:37.958Z [ERROR] (Thread-10) com.aws.greengrass.mqttclient.AwsIotMqtt5Client: Failed to connect to AWS IoT Core. {clientId=gg-5e37c73c40370bb47008-ggc-thing, error=socket operation timed out.}
```

Comment (MikeDombo) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: So if you only need 1, why does this step require 3?

Comment (bgklika) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: I assumed Nucleus made series of attempts before falling to disconnected state.

Comment (MikeDombo) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: No.

https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/3b555bb0321a78c4919ef59dc12ccaef7793b394/src/main/java/com/aws/greengrass/mqttclient/AwsIotMqtt5Client.java#L147-L162

Comment (bgklika) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: Ok, will change to 1

Comment (bgklika) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: onDisconnection() log another message - "Connection interrupted".
Message "Failed to connect to AWS IoT Core" we are waiting for logged from onConnectionFailure() which called 3 times after onDisconnection() .

But "Connection interrupted" is appeared already in log file.

Comment (MikeDombo) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: Yes, "Connection interrupted" means that the device was disconnected. If you see this, and there is no "Connection resumed" then it is offline.

Comment (bgklika) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1819]: Fixed in 224fb32b63284172f8b7ee71ec05037613d78680

Comment (bgklika) [reply uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature:1893]: At the moment has no steps to check log file miss some string.

**Review by MikeDombo — APPROVED** [review]

**Pull request merged**